src/blockrelay/graphene.cpp

```diff
@@ -225,13 +227,27 @@ bool CGrapheneBlockTx::HandleMessage(CDataStream &vRecv, CNode *pfrom)
     CGrapheneBlockTx grapheneBlockTx;
     vRecv >> grapheneBlockTx;
 
+    auto pblock = thinrelay.GetBlockToReconstruct(pfrom, grapheneBlockTx.blockhash);
+    if (pblock == nullptr)
+        return error("No block available to reconstruct for graphenetx");
+    DbgAssert(pblock->grapheneblock != nullptr, return false);
+
     // Message consistency checking
     CInv inv(MSG_GRAPHENEBLOCK, grapheneBlockTx.blockhash);
-    if (grapheneBlockTx.vMissingTx.empty() || grapheneBlockTx.blockhash.IsNull())
+    if (grapheneBlockTx.vMissingTx.empty())
+    {
+        // Normal effect if the IBLT decode on the other side completely failed
+        std::shared_ptr<CBlockThinRelay> backup = std::make_shared<CBlockThinRelay>(*pblock);
```

Comment:
I don't see how this is normal when there is a decode failure. When there is a decode failure, we should catch a runtime failure in `process` and then initiate failure recovery. We should not be rerequesting missing txs. Have you seen `grapheneBlockTx.vMissingTx.empty() == true` in testing?

Reply:
absolutely, I'll edit this message with details